### PR TITLE
[WIP] Added total level filtering to clan chat plugin

### PR DESCRIPTION
**Pull request by Biimmy**
cc filter

Added to existing Clan Chat plugin allowing for a minimum total level to be required by an account for the message to be shown.

Each account filtered is stored temporarily, until the plugin is disabled or the total level minimum variable in the config is changed.

Looking for way to not show message at all instead of altering it to say "FILTERED".

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatPlugin.java [hunk 1]
```diff
@@ -70,22 +97,132 @@ public void updateClanChatTitle()
...
+	 * @param message account name pulled from message and altered if neede
+	 */
+	private void lookupPlayer(final SetMessage message)
+	{
+		String account = message.getMessageNode().getName();
+
+		account = sanitize(account);
+
+		HiscoreSkill skill;
+		skill =  HiscoreSkill.valueOf(net.runelite.api.Skill.OVERALL.getName().toUpperCase());
```
Comment (detuks2) [inline runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatPlugin.java]: Move them in 1 line

File: runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatPlugin.java [hunk 2]
```diff
@@ -70,22 +97,132 @@ public void updateClanChatTitle()
...
+		message.getMessageNode().setName("FILTERED");
+		message.getMessageNode().setValue("");
+	}
+
+	/**
+	 * Looks through hashSet to see if the account has already been filtered.
+	 * @param name name of account to search
+	 * @return true if account has previously been filtered.
+	 */
+	private boolean isPlayerFiltered(String name)
```
Comment (detuks2) [inline runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatPlugin.java]: Use https://docs.oracle.com/javase/7/docs/api/java/util/HashSet.html#contains(java.lang.Object)

Check how hashSet works

File: runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatPlugin.java [hunk 3]
```diff
@@ -70,22 +97,132 @@ public void updateClanChatTitle()
...
+			{
+				return true;
+			}
 		}
+		return false;
 	}
 
 
-	private void insertClanRankIcon(final SetMessage message)
+	private String sanitize(String lookup)
```
Comment (detuks2) [inline runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatPlugin.java]: I am not sure what you try to do here, but if `message.getMessageNode().getName()` doesn't return clean name then should do changes there. 

You should write unit tests for this as well

File: runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatPlugin.java [hunk 4]
```diff
@@ -70,22 +97,132 @@ public void updateClanChatTitle()
...
+	 * @param setMessage message to be altered
+	 */
 	@Subscribe
 	public void onSetMessage(SetMessage setMessage)
 	{
+
+	    boolean ranked;
+	    //If filter minimum has changed, flush the stored accounts
+	    if (currentSetFilter != config.getTotalLevelFilter())
+		{
```
Comment (detuks2) [inline runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatPlugin.java]: Something off with style

File: runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatPlugin.java [hunk 5]
```diff
@@ -70,22 +97,132 @@ public void updateClanChatTitle()
...
+		}
+		try
+		{
+			SingleHiscoreSkillResult result = hiscoreClient.lookup(account, skill);
+			Skill hiscoreSkill = result.getSkill();
+
+			if (config.getTotalLevelFilter() > hiscoreSkill.getLevel())
+			{
+				//Add account to "blacklist" hashSet
+				filteredAccounts.add(message.getMessageNode().getName());
```
Comment (detuks2) [inline runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatPlugin.java]: Need to take care of concurrency. `filteredAccounts` need to be thread safe data structure

Comment (Biimmy) [reply runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatPlugin.java]: Going to take a look into ConncurrentHashMap for thread safe data as well as the ability to store TL along with Account Name as recommended above. Will update once I make some headway.

**Comment by Biimmy** [comment]
Closing this pull request due to hotfix released by Jagex May 14th 2018.
http://services.runescape.com/m=news/clan-chat-restrictions?oldschool=1

**Pull request closed without merging**